### DISCUSSION TITLE
Security check based on request's query parameter

### DISCUSSION THREAD

**Discussion by Asky-GH** [yahoo/elide#1985] [Elide 4]
Hi @aklish
Is there a possibility to get the request's query parameter to perform security checks?
For example, I have two classes:
```java
@Include(rootLevel = true, type = "children")
public class Child {
    @Id
    private UUID id;
    private UUID parentId;
}

@Include(rootLevel = true, type = "parents")
public class Parent {
    @Id
    private UUID id;
}
```
and when a user performs such a request - **children?filter=parentId==<id of some parent entity that doesn't belong to the user>**  I want to respond with 403...

**Comment by aklish** [comment]
The RequestScope implementation has the query params but the interface currently doesn't expose them:

https://github.com/yahoo/elide/blob/master/elide-core/src/main/java/com/yahoo/elide/core/RequestScope.java#L58

The interface ought to expose them.  This could be a quick 5 line PR.

**Reply by Asky-GH** [reply]
I don't quite understand why 5 lines?
Shouldn't we just add a method signature into the interface like so?
```java
public interface RequestScope {
    User getUser();
    Set<PersistentResource> getNewResources();
    Optional<MultivaluedMap<String, String>> getQueryParams();
}
```
Or you mean this dependency that should be added to **elide-annotations** module's pom?
```java
<dependency>
    <groupId>javax.ws.rs</groupId>
    <artifactId>javax.ws.rs-api</artifactId>
    <version>2.1.1</version>
    <scope>compile</scope>
</dependency>
```

**Reply by aklish** [reply]
Sorry for the confusion.  By five lines, I meant a very small PR (like 5ish).  Your change above is similar to what I was thinking.

Another possibility (without the pom change) is to do something like:

```java
    List<String> getQueryParameterByName(String parameterName);
```

This hides the underlying dependency on a javax ws Map and matches the interface for fetching headers:

```java
    String getRequestHeaderByName(String headerName);
```

**Reply by Asky-GH** [reply]
By the way, I've noticed that when using **OperationCheck** and querying some data using _filter_ like, for example, **books?filter=title==blabla** the check is being bypassed. I might be wrong but looks like a security flaw...

**Reply by aklish** [reply]
Are you referring to something like this?

https://github.com/yahoo/elide/security/advisories/GHSA-2mxr-89gf-rc4v

**Reply by Asky-GH** [reply]
Sorry, don't mind it. I was wrong.

**Reply by Brutus5000** [reply]
@aklish I have a related question: Why is Elide exposing just the security interface of the RequestScope? The life cycle hooks aren't just all about security, right? I am currently checking to put in some more business logic to it.

So I could either expand the security.RequestScope interface to whatever I need (I guess I also need query params) or swap the signature to use the "real" RequestScope.
